Source/EventFlow/EventStores/Files/FilesEventPersistence.cs

```diff
@@ -167,22 +168,25 @@ public class EventStoreLog
                     committedDomainEvents.Add(fileEventData);
                 }
 
-                using (var streamWriter = File.CreateText(_logFilePath))
+                //only allow a single write access to the _logFilePath file at one time
+                using (await _logFileSemaphore.WaitAsync(cancellationToken).ConfigureAwait(false))
```

Comment:
I don't think we need an additional semaphore: The whole method is already being locked by `_asyncLock`, making an inner lock redundant.